property.go

```diff
@@ -49,126 +34,130 @@ func (p RichTextProperty) GetType() PropertyType {
 	return p.Type
 }
 
-type PageTitleProperty struct {
-	ID    PropertyID   `json:"id,omitempty"`
-	Type  PropertyType `json:"type,omitempty"`
-	Title Paragraph    `json:"title"`
+type SelectProperty struct {
+	ID     ObjectID     `json:"id,omitempty"`
+	Type   PropertyType `json:"type"`
+	Select Option       `json:"select"`
 }
 
-func (p PageTitleProperty) GetType() PropertyType {
+func (p SelectProperty) GetType() PropertyType {
 	return p.Type
 }
 
-type FormatType string
-
-func (ft FormatType) String() string {
-	return string(ft)
-}
-
-type NumberProperty struct {
-	ID     ObjectID     `json:"id,omitempty"`
-	Type   PropertyType `json:"type"`
-	Format FormatType   `json:"format"`
+type MultiSelectProperty struct {
+	ID          ObjectID     `json:"id,omitempty"`
+	Type        PropertyType `json:"type"`
+	MultiSelect []Option     `json:"multi_select"`
 }
 
-func (p NumberProperty) GetType() PropertyType {
+func (p MultiSelectProperty) GetType() PropertyType {
 	return p.Type
 }
 
-type SelectProperty struct {
-	ID     ObjectID     `json:"id,omitempty"`
-	Type   PropertyType `json:"type"`
-	Select Select       `json:"select"`
+type Option struct {
+	ID    PropertyID `json:"id,omitempty"`
+	Name  string     `json:"name"`
+	Color Color      `json:"color,omitempty"`
 }
 
-func (p SelectProperty) GetType() PropertyType {
-	return p.Type
+type DateProperty struct {
+	ID   ObjectID     `json:"id,omitempty"`
+	Type PropertyType `json:"type"`
+	Date Date         `json:"date"`
 }
 
-type SelectOptionProperty struct {
-	ID     ObjectID     `json:"id,omitempty"`
-	Type   PropertyType `json:"type"`
-	Select Option       `json:"select"`
+type Date struct {
+	Start string `json:"start"`
+	End   string `json:"end"`
 }
 
-func (p SelectOptionProperty) GetType() PropertyType {
+func (p DateProperty) GetType() PropertyType {
 	return p.Type
 }
 
-type Select struct {
-	Options []Option `json:"options"`
+type FormulaProperty struct {
+	ID      ObjectID     `json:"id,omitempty"`
+	Type    PropertyType `json:"type"`
+	Formula Formula      `json:"formula"`
 }
 
-type MultiSelectProperty struct {
-	ID          ObjectID     `json:"id,omitempty"`
-	Type        PropertyType `json:"type"`
-	MultiSelect Select       `json:"multi_select"`
+type FormulaType string
+
+type Formula struct {
+	Type    FormulaType `json:"type"`
+	String  string      `json:"string,omitempty"`
+	Number  int         `json:"number,omitempty"`
+	Boolean bool        `json:"boolean,omitempty"`
+	Date    *Date       `json:"date,omitempty"`
 }
 
-func (p MultiSelectProperty) GetType() PropertyType {
+func (p FormulaProperty) GetType() PropertyType {
 	return p.Type
 }
 
-type MultiSelectOptionsProperty struct {
-	ID          ObjectID     `json:"id,omitempty"`
-	Type        PropertyType `json:"type"`
-	MultiSelect []Option     `json:"multi_select"`
+type RelationProperty struct {
+	ID       ObjectID     `json:"id,omitempty"`
+	Type     PropertyType `json:"type"`
+	Relation []PageID     `json:"relation"`
```

Comment:
That's incorrect. The right structure is the following according to the api doc, not just an array of strings.

```json
{
  "Project": {
    "relation": [
      {
        "id": "1d148a9e-783d-47a7-b3e8-2d9c34210355"
      }
    ]
  }
}
```